src/tespy/networks/networks.py

```diff
@@ -2523,6 +2525,178 @@ def postprocessing(self):
+            E_{\text{L}} = \sum_{conn=0}^N E_{\text{PH},conn}\\
+            E_{\text{D}} = E_{\text{F}} - E_{\text{P}} - E_{\text{L}}\\
+            \varepsilon = \frac{E_{\text{P}}}{E_{\text{F}}}
+
+            y_{\text{D},comp} =
+            \frac{\dot{E}_{\text{D},comp}}{\dot{E}_{\text{F}}}\\
+            y^*_{\text{D},comp} =
+            \frac{\dot{E}_{\text{D},comp}}{\dot{E}_{\text{D}}}
+        """
+        pamb_SI = self.convert_to_SI('p', pamb, self.p_unit)
+        Tamb_SI = self.convert_to_SI('T', Tamb, self.T_unit)
+
+        self.component_exergy_data = pd.DataFrame(
+            columns=['label', 'E_F', 'E_P', 'E_D', 'epsilon', 'y_Dk', 'y*_Dk'])
+
+        self.connection_exergy_data = pd.DataFrame(columns=['e_PH', 'E_PH'])
+
+        self.E_P = 0
+        self.E_F = 0
+        self.E_D = 0
+        self.E_L = 0
+
+        if len(E_F) == 0:
+            msg = ('Missing fuel exergy E_F of network.')
+            logging.error(msg)
+            raise hlp.TESPyNetworkError(msg)
+        elif len(E_P) == 0:
+            msg = ('Missing product exergy E_P of network.')
+            logging.error(msg)
+            raise hlp.TESPyNetworkError(msg)
+
+        # physical exergy of connections
+        for conn in self.conns.index:
+            conn.get_physical_exergy(pamb_SI, Tamb_SI)
+            self.connection_exergy_data.loc[conn.label] = [
+                conn.ex_physical, conn.Ex_physical]
+
+        # exergy balance of components
+
+        for cp in self.comps.index:
+            cp.exergy_balance()
+            if isinstance(cp, sink):
+                self.E_L += cp.E_D
+                cp_E_D = 0
+            else:
+                cp_E_D = cp.E_D
+
+            cp_E_F = cp.E_F
+            cp_E_P = cp.E_P
+            cp_epsilon = cp.epsilon
+            cp_on_num_busses = 0
+            for b in E_F + E_P + internal_busses:
+                if cp in b.comps.index:
+                    if cp_on_num_busses > 0:
+                        msg = (
+                            'The component ' + cp.label + ' is on multiple '
+                            'busses in the exergy analysis. Make sure that no '
+                            'component is connected to more than one of the '
+                            'busses passed to the exergy_analysis method.')
+                        logging.error(msg)
+                        raise hlp.TESPyNetworkError(msg)
+
+                    if b.comps.loc[cp, 'base'] == 'bus':
+                        cp_E_F = cp.E_F / cp.calc_bus_efficiency(b)
+                    else:
+                        cp_E_P = cp.E_P * cp.calc_bus_efficiency(b)
+                    cp_E_D = cp_E_F - cp_E_P
+                    cp_epsilon = cp_E_P / cp_E_F
+
+                    if b in E_F:
+                        if b.comps.loc[cp, 'base'] == 'bus':
+                            self.E_F += cp_E_F
+                        else:
+                            self.E_F -= cp_E_P
```

Comment:
Why is it cp_E_P and not cp_E_F ?

Reply:
Fuel exergy is inputs minus respective outputs (e.g. in a clausius rankine heat pump the compressor electrical power minus turbine electrical power). Therefore we need E_F for all power inputs and E_P for all power outputs and analogously for the E_P busses.